### ISSUE TITLE
Starting kong after setting KONG_LUA_CODE_CACHE returns empty reply from server

### ISSUE THREAD

**Issue by shiprabehera** [Mashape/kong-vagrant#46]
### Steps to Reproduce

1. `kong start`
2.  ` curl localhost:8001/apis`
This works.

3. `kong stop`
3. `export KONG_LUA_CODE_CACHE=false`
4. `kong start`
5.  ` curl localhost:8001/apis`

This returns `curl: (52) Empty reply from server`

Kong Version : 0.10.0

**Comment by Tieske** [comment]
That directive seems unreliable for now, as some other data that is supposed to persist also gets reloaded and hence the unpredictable results are returned.

I haven't yet figured it out where exactly this happens. Seems best for now to not turn code caching off. Make your changes and then issue a `bin/kong reload` or `bin/kong restart` command

**Comment by BeatMaster** [comment]
I'm experiencing the same issue using Vagrant with Centos 7 using version 0.10.1. Unfortunately, I've not had a chance to test against 0.10.2 to see if the issue has been resolved.

Seems to work OK with the LUA_CODE_CACHE switched on. From the investigation I did, it appears the issue is related to the defines within resty/worker/events.lua not persisting or being re-configured from init_worker().

Has there been any progress on resolving the issue?
